pkg/containerrun/containerrun.go

```diff
@@ -222,20 +270,35 @@ func handlePacket(
 
 	command := strings.TrimSpace(string(packet[:n]))
 	switch command {
-	case ProcessStart, ProcessStop:
+	case ProcessStart, ProcessStop, SignalQuit:
 		commands <- processCommand(command)
 	default:
 		// Bad commands are ignored. Else they could be used to DOS the runner.
 	}
 }
 
 func stopProcesses(processRegistry *ProcessRegistry, errors chan<- error) {
-	for _, err := range processRegistry.SignalAll(os.Kill) {
-		errors <- err
-	}
+	var wg sync.WaitGroup
```

Comment:
Oops, forgot to add: the error handling needs to be synchronous because it depends on the setting of the `active` boolean in the main loop. I did make the channel buffered at first, but then realized that that won't work because there is a (minimal) state machine involved, so the sequencing matters.

Reply:
Looking at the callers [[1]] [[2]], [[1]] is basically asynchronous and we should return as fast as possible.  In [[2]] we land in the [big loop](https://github.com/cloudfoundry-incubator/quarks-container-run/pull/10/files#diff-3d94c8020c09ed9fc87868a6d076b1ad4dc4d85e2c0d0d07949009a3517686aeR206) and wait for the process count to hit zero before doing anything.  So yeah, I think we're safe; we just need to remove the extraneous wait group.

It might be useful to document this behaviour, possibly as the doc comment for the parameter for `HandleSignals`?

[1]: https://github.com/cloudfoundry-incubator/quarks-container-run/pull/10/files#diff-3d94c8020c09ed9fc87868a6d076b1ad4dc4d85e2c0d0d07949009a3517686aeR173
[2]: https://github.com/cloudfoundry-incubator/quarks-container-run/pull/10/files#diff-3d94c8020c09ed9fc87868a6d076b1ad4dc4d85e2c0d0d07949009a3517686aeR666